Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -700,14 +701,21 @@ namespace AzToolsFramework
                 return AZ::Failure<AZStd::string>("Parent entity cannot be found while adding an entity.");
             }
 
-            // Get the alias of the parent entity in the owning template's DOM.
-            AZStd::string parentEntityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(parentId);
-            PrefabDomPath entityPathInOwningTemplate(parentEntityAliasPath.c_str());
-            
-            PrefabDom& owningTemplateDom =
-                m_prefabSystemComponentInterface->FindTemplateDom(owningInstanceOfParentEntity->get().GetTemplateId());
+            // Select the new entity (and deselect others).
+            AzToolsFramework::EntityIdList selection = { entityId };
+
+            SelectionCommand* selectionCommand = aznew SelectionCommand(selection, "");
+            selectionCommand->SetParent(undoBatch.GetUndoBatch());
+
+            ToolsApplicationRequests::Bus::Broadcast(&ToolsApplicationRequests::SetSelectedEntities, selection);
```

Comment:
what's the reason for moving this up here? Generally, selecting entities is one of the last things we do to prevent the selection being changed by actions below it.